### DISCUSSION TITLE
(15) My Precious [P.20220909]

### DISCUSSION THREAD

**Discussion by progettistamatto** [sapienzastudentsnetwork/basi-di-dati-2#32] [Progetti]
In questa discussion a tema il progetto [My Precious](https://elearning.uniroma1.it/pluginfile.php/1350504/mod_folder/content/0/%285%29%20P%20-%20Progetti/%2815%29%20My%20Precious%20-%20BD2.pdf), discuteremo di eventuali proposte di soluzione, dubbi o scelte progettuali in merito alla realizzazione del progetto in questione

**N.B.:** rispetto al testo oggetto dell'omonimo esame datato anch'esso 9 settembre 2022 le specifiche del progetto P.P.20220909 differiscono per l'aggiunta della seguente specifica:

- > Un’opera di proprietà del museo può essere presa in prestito da o data in prestito a un altro museo. Di un’opera in prestito interessa il museo di origine o destinazione, la data di inizio del prestito e quella di fine. Alcune opere, per via di un’importanza particolare o poiché troppo delicate, non possono essere date in prestito.

Per una precisazione alla **funzionalità n. 4** che il sistema deve offrire ai responsabili del patrimonio artistico e al management:

4. > calcolare il numero dei biglietti venduti in un dato giorno **per ognuna delle diverse tipologie**;

Per l'aggiunta di **3** ulteriori funzionalità richieste _(nuova n.5 - inserita tra la n.4 e la ex n.5, n.8 e n.9)_:

- > Dato un periodo di tempo, il numero medio di biglietti (di qualsivoglia tipologia) venduti (media giornaliera) nel periodo dato;

- > Dato un artista e un periodo di tempo, calcolare le opere realizzate dall’artista ed in possesso del museo che non sono state oggetto di restauri nel periodo dato;

- > Trovare le opere attualmente in restauro che non sono esposte da più tempo, insieme alla data di loro ultima esposizione.

Infine, la prima tipologia di biglietto viene nominata rispettivamente
- _full access_, nel progetto P.P.20220909
- _all-access_, nel testo oggetto dell'omonimo esame

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/Exyss/university-notes/raw/main/Bachelor/Secondo%20Anno/Basi%20di%20Dati%20II/My%20Precious.pdf) una soluzione proposta da Simone Bianco (Exyss)

> [!Warning]
_Le soluzioni di Simone Bianco fanno uso degli **Entity-Relationship (ER)** Diagrams su cui il corso si basava fino all'Anno Accademico 2023/2024, mentre **attualmente vengono utilizzati gli UML Class** Diagram. Sebbene i due siano estremamente simili, vi sono **alcune discordanze tra di loro**, in particolare l'**inversione di posizione dei vincoli di molteplicità agli estremi delle relationship**. Nonostante ciò, il rimanente delle fasi di analisi e progettazione delle soluzioni proposte risulta inaffetto da tale cambiamento. Per tanto, una volta comprese le differenze tra ER e Classi UML, le soluzioni risultano normalmente fruibili._

Rispondi a questo commento taggando @Exyss per eventuali dubbi o segnalazioni in merito

**Comment by IonutCicio** [comment]
Soluzione in UML [MyPrecious.pdf](https://github.com/user-attachments/files/16731816/MyPrecious.pdf)

Ho deciso di astrarre gli impieghi, entrare più nel dettaglio con il concetto di prestito e considerare (giustamente) le opere non di proprietà del museo, ed è venuta una soluzione abbastanza carina ✨

**Reply by Alby7503** [reply]
Non sarebbe meglio mettere “periodo” da “PresaInPrestito” direttamente in “Prestito”?

**Reply by IonutCicio** [reply]
@Alby7503 no, perché "DataInPrestito" ha già un suo periodo, quello d'impiego. Il periodo di "PresaInPrestito" e il periodo di "DataInPrestito" hanno due connotazioni diverse: quando un'opera è presa in prestito da un altro museo la puoi impiegare (in un'esposizione ad esempio) e "periodo" indica proprio il periodo in cui la puoi usare. Quando un'opera è data in prestito non puoi farci nulla, non è in tuo possesso durante quel periodo.

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/SimoneLid/BD2/raw/main/Progetti/My%20Precious/My%20Precious.pdf) una soluzione proposta da Simone Lidonnici (@SimoneLid)

**Comment by CosaFaSelect1** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/CasuFrost/university-notes-Informatica/raw/main/Secondo%20Anno/Secondo%20Semestre/Basi%20di%20Dati%202/Progetti/MyPrecious/MyPrecious.pdf) una soluzione proposta da Marco Casu
